CS483/CS483/Kartaclysm/Components/ComponentKartController.cpp

```diff
@@ -339,15 +403,30 @@ namespace Kartaclysm
 		m_fSpeed = fmaxf(m_fSpeed, m_fSpeed + (extra * (m_fSpeed / (m_fMaxSpeedStat * m_fSpeedScale))));
 	}
 
-	glm::quat ComponentKartController::GetRotationMinusSwerve()
+	void ComponentKartController::WheelieToggle()
 	{
-		m_pGameObject->GetTransform().SetRotation(glm::quat(glm::vec3(0.0f, m_fDirection, 0.0f)));
+		m_bWheelie = !m_bWheelie;
+	}
 
-		glm::quat result = m_pGameObject->GetTransform().GetRotation();
+	void ComponentKartController::Spinout(float p_fDuration)
+	{
+		m_fSpinout = fmaxf(p_fDuration * m_fDurabilityStat, m_fSpinout);
+	}
 
-		m_pGameObject->GetTransform().SetRotation(glm::quat(glm::vec3(0.0f, m_fDirection + m_fSwerve, 0.0f)));
+	void ComponentKartController::ArmorPlate(int p_iArmorStack)
+	{
+		// TODO: I think maybe there should only actually be 4 stacks *shrug*
+		m_iMaxSpeedCoreStat = 1 + p_iArmorStack;
+		m_iAccelerationCoreStat = 5 - p_iArmorStack;
+		m_iHandlingCoreStat = 5 - p_iArmorStack;
+		m_iDurabilityCoreStat = 1 + p_iArmorStack;
```

Comment:
As design lead, if you think it should be 4 stacks, go into the XML for the ability and change it. I even have the event sending you the MaxLayers parameter so why are you hardcoding it to be 5? David, ya dingus